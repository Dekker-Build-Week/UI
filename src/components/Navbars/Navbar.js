import React from "react";
import classNames from "classnames";
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";

import Button from "components/CustomButtons/Button.js";

import styles from "assets/jss/material-dashboard-react/components/headerStyle.js";

const useStyles = makeStyles(styles);

const logoImageUrl = "https://static.andigital.com/new-logos/ANDigital_LOGO_LONG_RGB_ON_WHITE.png";

export default function Header(props) {
  const classes = useStyles();

  const { color } = props;

  const appBarClasses = classNames({
    [" " + classes[color]]: color
  });

  return (
    <AppBar className={classes.appBar + appBarClasses}>
      <Toolbar className={classes.container}>
        <div className={classes.flex}>
          {/* Here we create navbar brand, based on route name */}
          <Button color="transparent" href="#" className={classes.title}>
            <img src = {logoImageUrl} alt = "Company Logo" className = {classes.logoImage} />
          </Button>
        </div>
      </Toolbar>
    </AppBar>
  );
}

Header.propTypes = {
  color: PropTypes.oneOf(["primary", "info", "success", "warning", "danger"]),
  handleDrawerToggle: PropTypes.func,
  routes: PropTypes.arrayOf(PropTypes.object)
};
