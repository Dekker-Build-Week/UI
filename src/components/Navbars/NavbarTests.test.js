import React from 'react';
import Navbar from './Navbar';

import renderer from 'react-test-renderer';

describe('Navbar Test Suite', () => {
    it('Renders the Navbar successfully', () => {
        const tree = renderer 
        .create(<Navbar><p>Hey</p></Navbar>)
            .toJSON();

        expect(tree).toMatchSnapshot();     
    })
})