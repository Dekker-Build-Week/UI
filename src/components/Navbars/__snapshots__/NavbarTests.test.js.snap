// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Test Suite Renders the Navbar successfully 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-1 MuiAppBar-colorPrimary mui-fixed"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular makeStyles-container-2 MuiToolbar-gutters"
  >
    <div
      className="makeStyles-flex-3"
    >
      <button
        className="MuiButtonBase-root MuiButton-root makeStyles-button-52 makeStyles-transparent-61 makeStyles-title-5 MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <img
            alt="Company Logo"
            className="makeStyles-logoImage-4"
            src="logo.png"
          />
        </span>
      </button>
    </div>
  </div>
</header>
`;
